ADALiOS/ADALiOS/ADKeychainTokenCache.m

```diff
@@ -387,35 +421,29 @@ - (void)addOrUpdateItem:(ADTokenCacheStoreItem *)item
                                            (id)kSecAttrAccessible : (id)kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly}];
         status = SecItemAdd((CFDictionaryRef)query, NULL);
         
-        [ADKeychainTokenCacheStore checkStatus:status details:@"Failed to add or update keychain entry." error:error];
+        if ([ADKeychainTokenCache checkStatus:status details:@"Failed to add or update keychain entry." error:error])
+        {
+            return NO;
+        }
     }
+    
+    return YES;
 }
 
-/*! Clears token cache details for specific keys.
- @param key: the key of the cache item. Key can be extracted from the ADTokenCacheStoreItem using
- the method 'extractKey'
- @param userId: The user for which the item will be removed. Can be nil, in which case items for all users with
- the specified key will be removed.
- The method does not raise an error, if the item is not found.
- */
-- (void)removeItemWithKey:(ADTokenCacheStoreKey *)key
-                   userId:(NSString *)userId
-                    error:(ADAuthenticationError * __autoreleasing *)error
+- (void)testRemoveAll:(ADAuthenticationError * __autoreleasing *)error
 {
-    if (!key)
+    @synchronized(self)
     {
-        ADAuthenticationError* adError = [ADAuthenticationError errorFromArgument:key argumentName:@"key"];
-        if (error)
+        NSMutableDictionary* query = [self queryDictionaryForKey:nil userId:nil additional:nil];
+        OSStatus status = SecItemDelete((CFDictionaryRef)query);
+        [ADKeychainTokenCache checkStatus:status details:@"Failed to remove all" error:error];
+        
+        NSArray* items = [self allItems:nil];
+        if ([items count])
         {
-            *error = adError;
+            NSLog(@"!!!!!!!!!!!!!!!!!!!! %lu items remaining...", (unsigned long)items.count);
```

Comment:
It's a test utility. The unit tests squash some of the logging and I wanted to make sure it got through.